### ISSUE TITLE
read-only editors for URI schemes project, std, lib should support LSP features

### ISSUE THREAD

**Issue by jurgenvinju** [usethesource/rascal-language-servers#35]
the current editor content provider for these schemes "work" in vscode, but the editors are "plaintext" only and do not offer syntax highlighting or reference resolution or an outline.

**Comment by jurgenvinju** [comment]
It looks like we have to implement the FileSystem abstractiojn for this? 

`class RascalFileSystemProvider implements vscode.FileSystemProvider`

that is doable, however several features of vscode.FileSystemProvider are not offered by the URIResolverRegistry API. For example creation time and file size are queries that are not supported, while we do have lastModification time. Also this vscode API supports file watchers, which would be a welcome addition to the URIResolverRegistry as well.

In other words: this would add features to all Rascal environments and could be worth it, but it is some days of work.

**Comment by jurgenvinju** [comment]
see https://github.com/microsoft/vscode/issues/73122

**Comment by jurgenvinju** [comment]
we implemented FileSystemProvider but the LSP does not work for this yet.

otherwise the FileSystemProvider works fine.

we should still add support to distinguish between writable and read-only schemes, such that we can make sure the editors for read-only schemes indeed provide read-only editors.

**Comment by DavyLandman** [comment]
They work now, and are also correctly marked as read only.

This was primarily enabled via: #158
